### DISCUSSION TITLE
R - bagaimana cara membuat pembulatan terbatas ke kelipatan 0.5?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#132] [Q&A] [answered]
Pada bahasa R, bagaimanakah cara membuat pembulatan terbatas ke kelipatan 0.5?
Misal, ada data: 
0.4, 
0.9,
1.6,
4.4

dibulatkan menjadi:
0.5,
1,
1.5,
4.5

**Comment by onlyphantom** [comment, marked as answer]
A simple solution is to multiply the numbers by 2, `round()` it, then divide by 2:

```
x <- c(0.4, 0.9, 1.6, 4.4)
round(x*2)/2
# output:
[1] 0.5 1.0 1.5 4.5
```

If performing on a dataframe column, change `x` accordingly:
```
round(mtcars$mpg*2)/2
# output:
 [1] 21.0 21.0 23.0 21.5 18.5 18.0 14.5 24.5 23.0 19.0 18.0 16.5 17.5 15.0 10.5 10.5
[17] 14.5 32.5 30.5 34.0 21.5 15.5 15.0 13.5 19.0 27.5 26.0 30.5 16.0 19.5 15.0 21.5
```

**Reply by Hertzspr** [reply]
Thank you.
